### PR TITLE
Combat overhaul - Baton specific changes

### PR DESCRIPTION
**Pull request by Gandalf2k15**
## About The Pull Request

This PR aims to force batons down the pecking order regarding combat, they have had their knockdown outright removed and the stamina damage they deal is 20.

We are not half-assing this combat overhaul.

## Changelog

:cl:
balance: All batons have had their knockdown effect nulled.
/:cl:

### CODE REVIEW

**Comment by MassacreDemigirl** [comment]
Im guessing the hit cooldown will be lowered considerably or else almost all batons become completely useless

**Comment by AzyreR** [comment]
So you WANT security to harmbaton, I fail to see how this is a wanted or good change at all, knockdown is literally all the batons have going for, and with the disarm getting removed on knockdown aswell.. I'm sorry but I can't wait to see sec get outplayed by a single esword and eshield unable to do anything and resort to lethals, which is the opposite of what i thought this server's aim was.

Like you geninely expect anyone from sec to land what.. 6 hits on someone only to put them in stamcrit while being put into softcrit themselves in 5 esword hits?

**Comment by RatFromTheJungle** [comment]
> So you WANT security to harmbaton, I fail to see how this is a wanted or good change at all, knockdown is literally all the batons have going for, and with the disarm getting removed on knockdown aswell.. I'm sorry but I can't wait to see sec get outplayed by a single esword and eshield unable to do anything and resort to lethals, which is the opposite of what i thought this server's aim was.
> 
> Like you geninely expect anyone from sec to land what.. 6 hits on someone only to put them in stamcrit while being put into softcrit themselves in 5 esword hits?

armor penning 20 damage

"The knockdown is all they have going for" ..

Full-armor penning 20 damage

The difference is one hit

probably less once you factor is sec's high melee armor

**Comment by AzyreR** [comment]
> > So you WANT security to harmbaton, I fail to see how this is a wanted or good change at all, knockdown is literally all the batons have going for, and with the disarm getting removed on knockdown aswell.. I'm sorry but I can't wait to see sec get outplayed by a single esword and eshield unable to do anything and resort to lethals, which is the opposite of what i thought this server's aim was.
> > Like you geninely expect anyone from sec to land what.. 6 hits on someone only to put them in stamcrit while being put into softcrit themselves in 5 esword hits?
> 
> armor penning 20 damage
> 
> "The knockdown is all they have going for" ..
> 
> Full-armor penning 20 damage
> 
> The difference is one hit
> 
> probably less once you factor is sec's high melee armor

You haven't taken into consideration that the batons stamina hit is about two time slower than a regular hit.

You haven't taken into consideration block chance.

Stamina damage can be recovered, easily fixable by using stims.

No way to knockdown, no way to slow down, no way to disarm, all you'd need to do is target their arms or legs with any weapon and you'll literally have them crawling unable to do anything but take hits.

If you need more of a visual representation, face to face, a naked antag with an esword against a armored officer with a baton, can just walk up to them, target their chest and tank every single baton hit the officer lands on them before they just kill the officer.

**Comment by RatFromTheJungle** [comment]
> > > So you WANT security to harmbaton, I fail to see how this is a wanted or good change at all, knockdown is literally all the batons have going for, and with the disarm getting removed on knockdown aswell.. I'm sorry but I can't wait to see sec get outplayed by a single esword and eshield unable to do anything and resort to lethals, which is the opposite of what i thought this server's aim was.
> > > Like you geninely expect anyone from sec to land what.. 6 hits on someone only to put them in stamcrit while being put into softcrit themselves in 5 esword hits?
> > 
> > 
> > armor penning 20 damage
> > "The knockdown is all they have going for" ..
> > Full-armor penning 20 damage
> > The difference is one hit
> > probably less once you factor is sec's high melee armor
> 
> You haven't taken into consideration that the batons stamina hit is about two time slower than a regular hit.
> 
> You haven't taken into consideration block chance.
> 
> Stamina damage can be recovered, easily fixable by using stims.
> 
> No way to knockdown, no way to slow down, no way to disarm, all you'd need to do is target their arms or legs with any weapon and you'll literally have them crawling unable to do anything but take hits.
> 
> If you need more of a visual representation, face to face, a naked antag with an esword against a armored officer with a baton, can just walk up to them, target their chest and tank every single baton hit the officer lands on them before they just kill the officer.

counter point: I take a 10u pill of sali-acid (can out heal most brute causing items) and face-tank the d-e-sword while spamming my baton or the sali-acid pen from the 300 cred brute first-aid kit at cargo

(it's also immensely more easy to get then dedicated anti-stamina items like bathsalts)

**Comment by RuneQuester** [comment]
> > > > So you WANT security to harmbaton, I fail to see how this is a wanted or good change at all, knockdown is literally all the batons have going for, and with the disarm getting removed on knockdown aswell.. I'm sorry but I can't wait to see sec get outplayed by a single esword and eshield unable to do anything and resort to lethals, which is the opposite of what i thought this server's aim was.
> > > > Like you geninely expect anyone from sec to land what.. 6 hits on someone only to put them in stamcrit while being put into softcrit themselves in 5 esword hits?
> > > 
> > > 
> > > armor penning 20 damage
> > > "The knockdown is all they have going for" ..
> > > Full-armor penning 20 damage
> > > The difference is one hit
> > > probably less once you factor is sec's high melee armor
> > 
> > 
> > You haven't taken into consideration that the batons stamina hit is about two time slower than a regular hit.
> > You haven't taken into consideration block chance.
> > Stamina damage can be recovered, easily fixable by using stims.
> > No way to knockdown, no way to slow down, no way to disarm, all you'd need to do is target their arms or legs with any weapon and you'll literally have them crawling unable to do anything but take hits.
> > If you need more of a visual representation, face to face, a naked antag with an esword against a armored officer with a baton, can just walk up to them, target their chest and tank every single baton hit the officer lands on them before they just kill the officer.
> 
> counter point: I take a 10u pill of sali-acid (can out heal most brute causing items) and face-tank the d-e-sword while spamming my baton or the sali-acid pen from the 300 cred brute first-aid kit at cargo
> 
> (it's also immensely more easy to get then dedicated anti-stamina items like bathsalts)

Im sorry, 10U sali-acid cannot face-tank a fucking double energy sword. 

Also, you just tried to turn this into a slapfight. That's not how combat works. Deswords also have block chance. Everything AzyreR said is still correct. Bath salts are not super hard to make, and even then, just make ephinedrine and becomes immune to stamina. Or some of the maint drugs.

**Comment by ChungusGamer666** [comment]
> So you WANT security to harmbaton, I fail to see how this is a wanted or good change at all, knockdown is literally all the batons have going for, and with the disarm getting removed on knockdown aswell.. I'm sorry but I can't wait to see sec get outplayed by a single esword and eshield unable to do anything and resort to lethals, which is the opposite of what i thought this server's aim was.
> 
> Like you geninely expect anyone from sec to land what.. 6 hits on someone only to put them in stamcrit while being put into softcrit themselves in 5 esword hits?

I'd rather get beaten to a pulp than stunned, because being beaten to a pulp at least lets me fight back.
Fuck stun combat.

File: modular_skyrat/modules/combat_overhaul/code/baton.dm [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+/obj/item/melee/baton/security
+	knockdown_time = (0 SECONDS)
+	stamina_damage = 20
+
+/obj/item/melee/baton/security/apply_stun_effect_end(mob/living/target)
+	return FALSE
+
+/obj/item/melee/baton/telescopic/contractor_baton
+	knockdown_time = (2.5 SECONDS)
```
Comment (GoldenAlpharex) [inline modular_skyrat/modules/combat_overhaul/code/baton.dm]: Those brackets are not necessary.
```suggestion
	knockdown_time = 0 SECONDS
	stamina_damage = 20

/obj/item/melee/baton/security
	knockdown_time = 0 SECONDS
	stamina_damage = 20

/obj/item/melee/baton/security/apply_stun_effect_end(mob/living/target)
	return FALSE

/obj/item/melee/baton/telescopic/contractor_baton
	knockdown_time = 2.5 SECONDS
```

**Review by GoldenAlpharex — CHANGES_REQUESTED** [review]

**Comment by Gandalf2k15** [comment]
I'll revisit this another year

**Pull request closed without merging**